OMEdit/OMEditLIB/Util/Utilities.cpp

```diff
@@ -976,19 +980,31 @@ QGenericMatrix<3,3, double> Utilities::getRotationMatrix(QGenericMatrix<3,1,doub
 QString Utilities::getGDBPath()
 {
 #if defined(_WIN32)
-#if defined(__MINGW32__) && !defined(__MINGW64__)
-  const char *sgdb = "/tools/msys/mingw32/bin/gdb.exe";
-#endif
-#if defined(__MINGW64__)
-  const char *sgdb = "/tools/msys/mingw64/bin/gdb.exe";
-#endif
   const char *OMDEV = getenv("OMDEV");
-  if (QString(OMDEV).isEmpty()) {
-    return QString(Helper::OpenModelicaHome).append(sgdb);
-  } else {
+  const char *OMDEV_MSYS = getenv("OMDEV_MSYS");
+  const char *MSYSTEM_PREFIX = getenv("MSYSTEM_PREFIX");
+  const char* msysEnv = System_openModelicaPlatform(); /* "ucrt64" or "mingw64" */
+
+  // MSYSTEM_PREFIX is set: <MSYSTEM_PREFIX>/bin/gdb.exe
+  if (!QString(MSYSTEM_PREFIX).isEmpty()) {
+    QString qMSYSTEM_PREFIX = QString(MSYSTEM_PREFIX).replace("\\", "/");
+    return QString(qMSYSTEM_PREFIX) + QString("/bin/gdb.exe");
+  }
+
+  // OMDEV_MSYS is set: <OMDEV_MSYS>/<CONFIG_OPENMODELICA_SPEC_PLATFORM>/bin/gdb.exe
+  if (!QString(OMDEV_MSYS).isEmpty()) {
+    QString qOMDEV_MSYS = QString(OMDEV_MSYS).replace("\\", "/");
+    return QString(qOMDEV_MSYS) + QString("/") + QString(msysEnv) + QString("/bin/gdb.exe");
+  }
+
+  // OMDEV is set: <OMDEV>/tools/msys/<CONFIG_OPENMODELICA_SPEC_PLATFORM>/bin/gdb.exe
+  if (!QString(OMDEV).isEmpty()) {
     QString qOMDEV = QString(OMDEV).replace("\\", "/");
-    return QString(qOMDEV).append(sgdb);
+    return QString(qOMDEV) + QString("/tools/msys/") + QString(msysEnv) + QString("/bin/gdb.exe");
```

Comment:
@adeas31 I think I made this too complicated and I'm not sure how to test if OMEdit can find gdb.
Any idea on how to simplify this or is this okay with you?

Reply:
Looks fine.

What is `MSYSTEM_PREFIX`? Did you consider the case that user has installed OpenModelica and doesn't have any `OMDEV` and `OMDEV_MSYS` defined.

You can test this by going to `Tools->Options->Debugger->Algorithmic Debugger`. `GDB Path` should show the value returned from this function.
Note: Makesure to remove the `GDBPath` value from `omedit.ini` file before starting OMEdit otherwise the settings value is preferred over the function value.

Reply:
`MSYSTEM_PREFIX` is defined by MSYS2 and points to the active environment, e.g. `C:\OMDev\tools\msys\ucrt64` or `C:\OMDev\tools\msys\mingw64`. So if OMEdit is run from that Shell (for developers mostly) that is available.

The default case (no MSYSTEM_PREFIX, no OMDEV_MSYS, no OMDEV):
```C
 // Default: <OPENMODELICAHOME>/tools/msys/<CONFIG_OPENMODELICA_SPEC_PLATFORM>/bin/gdb.exe
  return QString(Helper::OpenModelicaHome) + QString("/tools/msys/") + QString(msysEnv) + QString("bin/gdb.exe");
```

Reply:
GDB Path in `Tools->Options->Debugger->Algorithmic Debugger` looks correct on my machine (after removing omedit.ini).